Python/Product/TestAdapter/QualifiedFunctionNameWalker.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Python Tools for Visual Studio
+// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.PythonTools.Parsing.Ast;
+
+namespace Microsoft.PythonTools.Analysis {
+    /// <summary>
+    /// Computes the fully qualified function name, including name of the enclosing class for methods,
+    /// and, recursively, names of any outer functions.
+    /// </summary>
+    /// <example>
+    /// Given this code:
+    /// <code>
+    /// class A:
+    ///   def b(self):
+    ///     def c():
+    ///       class D:
+    ///         def e(self):
+    ///           pass
+    /// </code>
+    /// And with the current statement being <c>pass</c>, the qualified name is "D.e in c in A.b".
+    /// </example>
+    class QualifiedFunctionNameWalker : PythonWalker {
```

Comment:
This is actually a rename, GitHub just doesn't show it properly.

Reply:
There are 2 copies of it now, in TestAdapter and Debugger projects. We could have the Debugger project Add as Link instead to avoid the copy. Note: Legacy debugger, which needs this, will be removed in the future

Reply:
Ok